### PR TITLE
Convert feed content fetcher, announcements fetcher, on this day fetcher away from AFNetworking

### PR DESCRIPTION
**Pull request by joewalsh**

### CODE REVIEW

File: WMF Framework/WMFAnnouncementsFetcher.m [hunk 1]
```diff
@@ -41,31 +28,48 @@ - (void)fetchAnnouncementsForURL:(NSURL *)siteURL force:(BOOL)force failure:(WMF
...
+        
+        NSError *mantleError = nil;
+        NSArray<WMFAnnouncement *> *announcements = [MTLJSONAdapter modelsOfClass:[WMFAnnouncement class] fromJSONArray:announcementJSONs error:&mantleError];
+        if (mantleError){
+            failure(mantleError);
+            return;
+        }
+        
+        WMFAnnouncement *announcement = announcements.firstObject;
+        if (![announcement isKindOfClass:[WMFAnnouncement class]]) {
```
[the remark this answers is not in the document]
Comment (joewalsh) [reply WMF Framework/WMFAnnouncementsFetcher.m]: I'm not even sure this check is necessary, I'd assume we'd get an error from mantle if it was unable to serialize the objects to the right class

File: WMF Framework/WMFOnThisDayEventsFetcher.m [hunk 2]
```diff
@@ -54,28 +41,38 @@ - (void)fetchOnThisDayEventsForURL:(NSURL *)siteURL month:(NSUInteger)month day:
...
+        
+        NSError *mantleError = nil;
+        NSArray<WMFFeedOnThisDayEvent *> *events = [MTLJSONAdapter modelsOfClass:[WMFFeedOnThisDayEvent class] fromJSONArray:eventJSONs error:&mantleError];
+        if (mantleError) {
+            failure([NSError wmf_errorWithType:WMFErrorTypeUnexpectedResponseType userInfo:nil]);
+            return;
+        }
+        
+        WMFFeedOnThisDayEvent *event = events.firstObject;
+        if (![event isKindOfClass:[WMFFeedOnThisDayEvent class]]) {
```
Comment (nambatee) [inline WMF Framework/WMFOnThisDayEventsFetcher.m]: same question as above with annoucements

File: Wikipedia/Code/Session.swift [hunk 3]
```diff
@@ -287,26 +287,38 @@ import Foundation
...
             }
         })
     }
     
+    @objc(getJSONDictionaryFromURL:ignoreCache:completionHandler:) public func getJSONDictionary(from url: URL?, ignoreCache: Bool, completionHandler: @escaping ([String: Any]?, HTTPURLResponse?, Error?) -> Swift.Void) {
+        guard let url = url else {
+            completionHandler(nil, nil, NSError.wmf_error(with: .invalidRequestParameters))
+            return
+        }
+        var request = URLRequest(url: url)
```
Comment (nambatee) [inline Wikipedia/Code/Session.swift]: should this be using our own `request` from line `138` or some other wrapper that sets the `"User-Agent"` and `Accept` headers?

File: WMF Framework/WMFAnnouncementsFetcher.m (hunk 1 above)
[the remark this answers is not in the document]
Comment (nambatee) [reply WMF Framework/WMFAnnouncementsFetcher.m]: when I set `announcementsJSONs` to `@[@"foo", @"bar"]`, it crashes with `NSInvalidArgumentException`

[the remark this answers is not in the document]
Comment (joewalsh) [reply WMF Framework/WMFAnnouncementsFetcher.m]: 🤦‍♂️

**Pull request merged**